plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RewriteCast.java

```diff
@@ -66,6 +75,11 @@ private boolean pushdownSupported(JdbcTypeHandle sourceType, Type targetType)
         return switch (targetType) {
             case SmallintType _, IntegerType _, BigintType _ ->
                     SUPPORTED_SOURCE_TYPE_FOR_INTEGRAL_CAST.contains(sourceType.jdbcType());
+            // varchar -> char is unsupported as varchar supports multi-byte characters whereas char supports only single byte characters.
+            case CharType _ -> CHAR == sourceType.jdbcType();
+            // char -> varchar is not supported as Redshift doesn't pad char value with blanks whereas Trino pads char value with blanks.
+            // cast to unbounded varchar is unsupported as Redshift doesn't support unbounded varchar
+            case VarcharType varcharType -> VARCHAR == sourceType.jdbcType() && !varcharType.isUnbounded() && varcharType.getLength().orElse(REDSHIFT_MAX_VARCHAR) <= REDSHIFT_MAX_VARCHAR;
```

Comment:
nit: If varchar is bounded then length will always exists - How about `VarcharType#getBoundedLength`

Reply:
Agreed. modified.